Gems/Atom/Tools/ShaderManagementConsole/Scripts/GenerateShaderVariantListForMaterials.py

```diff
@@ -39,79 +39,110 @@ def main():
+        variantInfo.stableId = stableId
+        options = {}
+
+        shaderOptionDescriptors = shaderOptionGroup.GetShaderOptionDescriptors()
+        for shaderOptionDescriptor in shaderOptionDescriptors:
+            optionName = shaderOptionDescriptor.GetName()
+            optionValue = shaderOptionGroup.GetValueByOptionName(optionName)
+            if not optionValue.IsValid():
+                continue
+
+            valueName = shaderOptionDescriptor.GetValueName(optionValue)
+            options[optionName.ToString()] = valueName.ToString()
+
+        if len(options) != 0:
+            variantInfo.options = options
+            shaderVariants.append(variantInfo)
+            stableId += 1
+
+    shaderVariantList.shaderVariants = shaderVariants
 
-    # generate the default save file path by replacing the extension of the open shader file
-    pre, ext = os.path.splitext(filename)
-    defaultShaderVariantListFilePath = f'{pre}.shadervariantlist'
-    
     # clean previously generated shader variant list file so they don't clash.
-    pre, ext = os.path.splitext(shaderVariantList.shaderFilePath)
+    pre, ext = os.path.splitext(shaderAssetInfo.relativePath)
     projectShaderVariantListFilePath = os.path.join(azlmbr.paths.projectroot, PROJECT_SHADER_VARIANTS_FOLDER, f'{pre}.shadervariantlist')
-    
+    pre, ext = os.path.splitext(filename)
+    defaultShaderVariantListFilePath = f'{pre}.shadervariantlist'
+
     clean_existing_shadervariantlist_files([
         projectShaderVariantListFilePath
     ])
 
-    # Save the shader variant list file
-    if is_save_in_project_folder:
-        shaderVariantListFilePath = projectShaderVariantListFilePath
-    else:
-        shaderVariantListFilePath = defaultShaderVariantListFilePath
-
-    shaderVariantListFilePath = shaderVariantListFilePath.replace("\\", "/")
+    defaultShaderVariantListFilePath = defaultShaderVariantListFilePath.replace("\\", "/")
 
-    # Save the document in shader management console
-    saveResult = azlmbr.atomtools.AtomToolsDocumentSystemRequestBus(
-        azlmbr.bus.Broadcast,
-        'SaveDocumentAsChild',
+    # Open default shader variant list document
+    documentId = azlmbr.atomtools.AtomToolsDocumentSystemRequestBus(
+       azlmbr.bus.Broadcast,
+       'OpenDocument',
+       defaultShaderVariantListFilePath
+    )
+    # Update shader variant list
+    azlmbr.shadermanagementconsole.ShaderManagementConsoleDocumentRequestBus(
+        azlmbr.bus.Event,
+        'SetShaderVariantListSourceData',
         documentId,
-        shaderVariantListFilePath
+        shaderVariantList
     )
-
-    if saveResult:
-        msgBox = QtWidgets.QMessageBox(
-            QtWidgets.QMessageBox.Information,
-            "Shader Variant List File Successfully Generated",
-            f".shadervariantlist file was saved in:\n{shaderVariantListFilePath}",
-            QtWidgets.QMessageBox.Ok
-        )
-        msgBox.exec()
-
+    
```

Comment:
I don't see the data from the shader variant list being saved anywhere. If you plan for the scripts to still be written in terms of the document class, then I would leave the functions that were moved back on the document class and its bus. Have the script create a new document, do all of this work to build the shader variant list, set the shader variant list on the document, then save the shader variant list to the desired location.

Some things that you'll need to look out for are making sure that the application shuts down cleanly. When I ported this to C++ originally, there were asset and named dictionary leaks because the script was holding on to references

Reply:
Since the file are only allow to save in 2 locations, I think its reasonable to just update the default .shadervariantlist file and exit the script in an unsaved state, user can save manually or close the application without saving (there will be a warning popup if the data is not save). 

Regarding the leaks, I did find a lot of them when I close the program. I can confirmed the call to `GetMaterialInstanceShaderItems` will result in NameDictionary leaked, maybe we need to update the function somehow?

Reply:
It would be nice if we could do a NewDocument command instead of OpenDocument, so the script doesn't assume where the user will want to save the file. I agree with the idea of just updating the shader variant list, and don't save. The user can save if they want, or discard it, as you described.